test/helpers/versions.spec.js

```diff
@@ -58,6 +58,223 @@ describe('helpers/versions', () => {
+    });
+    it('upgrades .x minor ranges', () => {
+      const upgradeVersions = [
+        {
+          newVersion: '1.4.1',
+          newVersionMajor: 1,
+          upgradeType: 'minor',
+          changeLogFromVersion: '1.3.0',
+          changeLogToVersion: '1.4.1',
+        },
+      ];
+      expect(versionsHelper.determineUpgrades(qJson, '1.3.x', defaultConfig)).toEqual(upgradeVersions);
+    });
+    it('upgrades tilde ranges without pinning', () => {
+      const upgradeVersions = [
+        {
+          newVersion: '~1.4.0',
+          newVersionMajor: 1,
+          upgradeType: 'minor',
+          changeLogFromVersion: '1.3.0',
+          changeLogToVersion: '1.4.1',
+          isRange: true,
+        },
+      ];
+      const config = Object.assign({}, defaultConfig);
+      config.pinVersions = false;
+      expect(versionsHelper.determineUpgrades(qJson, '~1.3.0', config)).toEqual(upgradeVersions);
+    });
+    it('upgrades .x major ranges without pinning', () => {
+      const upgradeVersions = [
+        {
+          newVersion: '1.x',
+          newVersionMajor: 1,
+          upgradeType: 'major',
+          changeLogFromVersion: '0.9.7',
+          changeLogToVersion: '1.4.1',
+          isRange: true,
+        },
+      ];
+      const config = Object.assign({}, defaultConfig);
+      config.pinVersions = false;
+      expect(versionsHelper.determineUpgrades(qJson, '0.x', config)).toEqual(upgradeVersions);
+    });
+    it('upgrades .x minor ranges without pinning', () => {
+      const upgradeVersions = [
+        {
+          newVersion: '1.4.x',
+          newVersionMajor: 1,
+          upgradeType: 'minor',
+          changeLogFromVersion: '1.3.0',
+          changeLogToVersion: '1.4.1',
+          isRange: true,
+        },
+      ];
+      const config = Object.assign({}, defaultConfig);
+      config.pinVersions = false;
+      expect(versionsHelper.determineUpgrades(qJson, '1.3.x', config)).toEqual(upgradeVersions);
+    });
+    it('upgrades shorthand major ranges without pinning', () => {
+      const upgradeVersions = [
+        {
+          newVersion: '1',
+          newVersionMajor: 1,
+          upgradeType: 'major',
+          changeLogFromVersion: '0.9.7',
+          changeLogToVersion: '1.4.1',
+          isRange: true,
+        },
+      ];
+      const config = Object.assign({}, defaultConfig);
+      config.pinVersions = false;
+      expect(versionsHelper.determineUpgrades(qJson, '0', config)).toEqual(upgradeVersions);
+    });
+    it('upgrades shorthand minor ranges without pinning', () => {
+      const upgradeVersions = [
+        {
+          newVersion: '1.4',
+          newVersionMajor: 1,
+          upgradeType: 'minor',
+          changeLogFromVersion: '1.3.0',
+          changeLogToVersion: '1.4.1',
+          isRange: true,
+        },
+      ];
+      const config = Object.assign({}, defaultConfig);
+      config.pinVersions = false;
```

Comment:
Something I realized later was that this is another case where `pinVersions` can be passed as a property in an object passed to `Object.assign`.